### ISSUE TITLE
Wrong Plugin Being Downloaded

### ISSUE THREAD

**Issue by WebTechGlobal** [TGMPA/TGM-Plugin-Activation#491]
I have the class working within WP Multitask plugin - slight adaption of class required. Mainly where "theme.php" is used.

When bulk installing a strange problem occurs. This is a copy and paste below showing an incorrect plugin being installed (URL does not match plugin name). I'm still getting to know the inner workings of the class so I've no ideas why this is happening but I'll try my best to figure it out. The fact that the incorrect URL appears in the third plugin might tell us something.

"The installation process is starting. This process may take a while on some hosts, so please be patient.

Installing Plugin Blip TV Episodes Widget (1/3)

Installing Plugin Star Citizen Fan Site Kit Beta (2/3)
Downloading install package from https://downloads.word
![bulk plugin activation class installing wrong plugin error](https://cloud.githubusercontent.com/assets/3390050/10775551/412aa58e-7d01-11e5-8e02-a8173bcce04d.png)
press.org/plugin/background-videos-for-visual-composer.1.0.5.zip…
Unpacking the package…
Installing the plugin…
Plugin install failed.

Installing Plugin WTG Notice Boxes (3/3)
Downloading install package from https://downloads.wordpress.org/plugin/background-videos-for-visual-composer.1.0.5.zip…
Unpacking the package…
Installing the plugin…
Plugin install failed.
All installations have been completed."

**Comment by GaryJones** [comment]
Thanks for reporting @WebTechGlobal.

Please can you restore the previous plugin versions, and try the same again, but with a stock version of TGMPA, instead of something you've customised?

Saying "I've adapted the script, and now something is broken" is what it appears you've said so far.

**Comment by WebTechGlobal** [comment]
The problem hasn't happened with the latest version. I accidentally switched to an older version during testing.

It's still an issue because nothing I changed could cause what I saw happening. All I changed was the arguments regarding theme or admin (plugin) so that I could display the table within my own page. The data I was working with was a plugin data call on WordPress.org API based on a specific author, me. I was feeding the results of the call directly into this class. So I know the plugins involved, I've examined the entire results and it only contains my own plugins.

The plugins this class was displaying during bulk processing were totally new to me. 

It's just something to be aware of should it come up again.
